### PR TITLE
Edit Team Name

### PR DESCRIPTION
**Pull request by samisuteria**
@aabidshamji cleaned up unneeded changes from: https://github.com/AgaKhanFoundation/WCF-iOS/pull/337

### CODE REVIEW

File: Steps4Impact.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (compnerd) [inline Steps4Impact.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: Seems unrelated?

File: Steps4Impact/Networking/AKFCausesService.swift [hunk 2]
```diff
@@ -149,7 +149,10 @@ class AKFCausesService: Service {
     shared.request(.patch, endpoint: .participant(fbId: fbid),
                    parameters: JSON(["team_id": "null"]), completion: completion)
   }
-
+  static func editTeamName(team: Int, name: String, completion: ServiceRequestCompletion? = nil) {
```
Comment (compnerd) [inline Steps4Impact/Networking/AKFCausesService.swift]: I think that renaming `team` to `teamId` is a good idea.

File: Steps4Impact/Settings/TeamSettingsViewController.swift [hunk 3]
```diff
@@ -71,11 +71,34 @@ extension TeamSettingsViewController: SettingsActionCellDelegate {
...
           string: Strings.Share.item)
+    case .editname:
+      AKFCausesService.getParticipant(fbid: Facebook.id) { (result) in
+        guard let participant = Participant(json: result.response) else { return }
+        guard participant.team != nil else { return }
+        let alert = UIAlertController(title: Strings.TeamSettings.editTeamName,
+                                      message: Strings.TeamSettings.editTeamNameMessage,
+                                      preferredStyle: UIAlertController.Style.alert)
+        let action = UIAlertAction(title: Strings.TeamSettings.update, style: .default) { (_) in
+          let textField = alert.textFields![0] as UITextField
```
Comment (samisuteria) [inline Steps4Impact/Settings/TeamSettingsViewController.swift]: shouldn't use force unwrap

File: Steps4Impact/Networking/AKFCausesService.swift (hunk 2 above)
Comment (compnerd) [inline Steps4Impact/Networking/AKFCausesService.swift]: Should this be `editTeamName` or `setTeamName` perhaps?  This allows you to set the team name, which is the "edit"

File: Steps4Impact/Settings/TeamSettingsViewController.swift [hunk 4]
```diff
@@ -71,11 +71,34 @@ extension TeamSettingsViewController: SettingsActionCellDelegate {
           viewController: self,
           shareButton: button,
           string: Strings.Share.item)
+    case .editname:
+      AKFCausesService.getParticipant(fbid: Facebook.id) { (result) in
+        guard let participant = Participant(json: result.response) else { return }
+        guard participant.team != nil else { return }
```
Comment (compnerd) [inline Steps4Impact/Settings/TeamSettingsViewController.swift]: Use a `guard let team = participant.team else { return }`

File: Steps4Impact/Settings/TeamSettingsViewController.swift [hunk 5]
```diff
@@ -71,11 +71,34 @@ extension TeamSettingsViewController: SettingsActionCellDelegate {
...
+        let alert = UIAlertController(title: Strings.TeamSettings.editTeamName,
+                                      message: Strings.TeamSettings.editTeamNameMessage,
+                                      preferredStyle: UIAlertController.Style.alert)
+        let action = UIAlertAction(title: Strings.TeamSettings.update, style: .default) { (_) in
+          let textField = alert.textFields![0] as UITextField
+          let newName = textField.text!
+          AKFCausesService.editTeamName(team: (participant.team?.id)!, name: newName) { (_) in
```
Comment (compnerd) [inline Steps4Impact/Settings/TeamSettingsViewController.swift]: This won't need to be forced/unwrapped with the `guard let`.

File: Steps4Impact/Settings/TeamSettingsViewController.swift [hunk 6]
```diff
@@ -71,11 +71,34 @@ extension TeamSettingsViewController: SettingsActionCellDelegate {
...
+    case .editname:
+      AKFCausesService.getParticipant(fbid: Facebook.id) { (result) in
+        guard let participant = Participant(json: result.response) else { return }
+        guard participant.team != nil else { return }
+        let alert = UIAlertController(title: Strings.TeamSettings.editTeamName,
+                                      message: Strings.TeamSettings.editTeamNameMessage,
+                                      preferredStyle: UIAlertController.Style.alert)
+        let action = UIAlertAction(title: Strings.TeamSettings.update, style: .default) { (_) in
+          let textField = alert.textFields![0] as UITextField
+          let newName = textField.text!
```
Comment (compnerd) [inline Steps4Impact/Settings/TeamSettingsViewController.swift]: mmm...perhaps worth ensuring that the text is non-nil?

File: Steps4Impact/Settings/TeamSettingsViewController.swift [hunk 7]
```diff
@@ -71,11 +71,34 @@ extension TeamSettingsViewController: SettingsActionCellDelegate {
...
           shareButton: button,
           string: Strings.Share.item)
+    case .editname:
+      AKFCausesService.getParticipant(fbid: Facebook.id) { (result) in
+        guard let participant = Participant(json: result.response) else { return }
+        guard participant.team != nil else { return }
+        let alert = UIAlertController(title: Strings.TeamSettings.editTeamName,
+                                      message: Strings.TeamSettings.editTeamNameMessage,
+                                      preferredStyle: UIAlertController.Style.alert)
+        let action = UIAlertAction(title: Strings.TeamSettings.update, style: .default) { (_) in
```
Comment (compnerd) [inline Steps4Impact/Settings/TeamSettingsViewController.swift]: Not sure if it helps outlining this.  Why not just put this inline in the `addAction`?

File: Steps4Impact/Settings/TeamSettingsViewController.swift [hunk 8]
```diff
@@ -71,11 +71,34 @@ extension TeamSettingsViewController: SettingsActionCellDelegate {
...
+          AKFCausesService.editTeamName(team: (participant.team?.id)!, name: newName) { (_) in
+            onMain {
+              alert.dismiss(animated: true, completion: nil)
+              self.navigationController?.popViewController(animated: true)
+              NotificationCenter.default.post(name: .teamChanged, object: nil)
+            }
+          }
+        }
+        alert.addTextField { (textField) in
+          textField.text = "Enter Team Name"
```
Comment (compnerd) [inline Steps4Impact/Settings/TeamSettingsViewController.swift]: Probably worth making a translatable string for this

File: Steps4Impact/Settings/TeamSettingsViewController.swift [hunk 9]
```diff
@@ -71,11 +71,34 @@ extension TeamSettingsViewController: SettingsActionCellDelegate {
...
+        }
+        alert.addAction(action)
+        UIApplication.shared.keyWindow?.rootViewController?.present(alert, animated: true, completion: nil)
```
Comment (compnerd) [inline Steps4Impact/Settings/TeamSettingsViewController.swift]: Why is this being presented from the root view controller rather than this view controller?

**Review by compnerd — CHANGES_REQUESTED** [review]

File: Steps4Impact/Networking/AKFCausesService.swift (hunk 2 above)
Comment (samisuteria) [reply Steps4Impact/Networking/AKFCausesService.swift]: I think going with edit is fine - set implies it wasn't set before - which it was during team creation

**Comment by samisuteria** [comment]
Updated PR based on feedback. 

Simulator Screen Shot - iPhone Xʀ - 2019-11-08 at 18 43 05

File: Steps4Impact/Supporting Files/en.lproj/Localizable.strings [hunk 10]
```diff
@@ -126,8 +126,8 @@
 "TeamSettings.done" = "Done";
 "TeamSettings.update" = "Update";
 "TeamSettings.leave" = "Leave Team?";
-"TeamSettings.editTeamName" = "Edit Team Name";
-"TeamSettings.editTeamNameMessage" = "Please enter a new team name";
+"TeamSettings.editTeamName" = "Edit Team Name ";
+"TeamSettings.editTeamNameMessage" = "Please enter a new team name ";
```
Comment (compnerd) [inline Steps4Impact/Supporting Files/en.lproj/Localizable.strings]: Why the trailing whitespace?

**Review by compnerd — DISMISSED** [review]
